packages/common/infra/src/sync/indexer/impl/indexeddb/match.ts

```diff
@@ -0,0 +1,126 @@
+export class Match {
+  scores = new Map<number, number>();
+  highlighters = new Map<
```

Comment:
match is where the intermediate state of the query. If the name is not appropriate, do you have any suggestions?

Reply:
the highlighter here is ultimately ref to the highlight method in FullTextInvertedIndex. It may be more appropriate to replace it with the ranges array? `[field, start, end][]`

Reply:
> It may be more appropriate to replace it with the ranges array? [field, start, end][]
Agreed